### PR TITLE
Fix issue when doing multiple artifact packaging calls in a raw

### PR DESCRIPTION
**Pull request by purbon**
This PR makes sure the start situation where each packaging call starts is the same (with an Gemfile without plugins), so there known issue with bootstrapping is not present. There were missing jars because we call the ```rake artifact:prepare``` multiple times, having the first starting with an Gemfile without plugins while the other ones had plugins, so the bootstrap got the jars wrong.

It also relocated defaults files to the pkg dir, so they are not polluting the root directory.

This PR aims to fix the root cause of https://github.com/elasticsearch/logstash/issues/2782  and the missing jars from https://github.com/elasticsearch/logstash/issues/2780

### CODE REVIEW

**Comment by purbon** [comment]
@colinsurprenant @ph would you guys be nice to review this?

File: rakelib/artifacts.rake [hunk 1]
```diff
@@ -43,17 +43,24 @@ namespace "artifact" do
...
   task "freeze-defaults-gemfile" => ["bootstrap", "plugin:install-default"] do
-    FileUtils.cp("Gemfile", "Gemfile.defaults")
-    FileUtils.cp("Gemfile.jruby-1.9.lock", "Gemfile.jruby-1.9.lock.defaults")
+    base_dir = "pkg/gemfile/defaults"
+    FileUtils.cp("Gemfile", "#{base_dir}/Gemfile.defaults")
+    FileUtils.cp("Gemfile.jruby-1.9.lock", "#{base_dir}/Gemfile.jruby-1.9.lock.defaults")
   end
 
-  task "prepare" => ["bootstrap", "use-defaults-gemfile", "plugin:install-default"]
+  task "prepare" => ["use-core-gemfile","bootstrap", "use-defaults-gemfile", "plugin:install-default"]
```
Comment (jsvd) [inline rakelib/artifacts.rake]: maybe "set-base-gemfile" instead?

Comment (purbon) [inline rakelib/artifacts.rake]: can do.

File: rakelib/artifacts.rake [hunk 2]
```diff
@@ -43,17 +43,24 @@ namespace "artifact" do
...
-    FileUtils.cp("Gemfile.jruby-1.9.lock.defaults", "Gemfile.jruby-1.9.lock")
+    base_dir = "pkg/gemfile/defaults"
+    FileUtils.cp("#{base_dir}/Gemfile.defaults", "Gemfile")
+    FileUtils.cp("#{base_dir}/Gemfile.jruby-1.9.lock.defaults", "Gemfile.jruby-1.9.lock")
   end
 
   task "freeze-defaults-gemfile" => ["bootstrap", "plugin:install-default"] do
-    FileUtils.cp("Gemfile", "Gemfile.defaults")
-    FileUtils.cp("Gemfile.jruby-1.9.lock", "Gemfile.jruby-1.9.lock.defaults")
+    base_dir = "pkg/gemfile/defaults"
```
Comment (jsvd) [inline rakelib/artifacts.rake]: Also this base_dir var could be moved to a constant DEFAULT_GEMFILE_DIR or similar

**Comment by purbon** [comment]
@jsvd update your comments, thanks.

**Comment by jsvd** [comment]
haven't tested yet. testing

**Comment by purbon** [comment]
thanks man, I know takes time to test.

**Comment by purbon** [comment]
doing 

```
rake artifact:tar
rake artifact:deb
```
and then checking if the deb package works with ES, I get no error.

**Comment by purbon** [comment]
This PR is no longer valid as we change some of the things involving this.

**Pull request closed without merging**